### DISCUSSION TITLE
Synchronize database after connection.

### DISCUSSION THREAD

**Discussion by d-kurchenko** [prisma/prisma#18522] [Q&A]
In typeorm, when creating a database connection, I can use the **synchronize** parameter and then orm will check the database with my entities and change the database on the fly. 
```
new DataSource({
  synchronize: true,
  logging: false,
  type: "better-sqlite3",
  ...
});
```
Is it possible to implement this behavior using Prisma?

**Comment by ludralph** [comment]
Hi @pvmnsd 👋 

Prisma doesn't have the exact synchronize behaviour but i think `Prisma Migrate` is somewhat similar. Prisma Migrate allows you to define database schema changes using the Prisma schema language and apply those changes to your database. It can also generate SQL migration scripts that you can review and apply manually. See the [documentation](https://www.prisma.io/docs/concepts/components/prisma-migrate) for more information on Prisma Migrate

**Reply by nikolasburk** [reply]
@pvmnsd you might also be interested in the [`prisma db push`](https://www.prisma.io/docs/reference/api-reference/command-reference#db-push) command which I think gets closest to the behaviour of TypeORM's `synchronize` because it just synchronizes the state of your models with the DB schema without needing to take care of migrations.

**Reply by d-kurchenko** [reply]
@ludralph , @nikolasburk  my app using multiple databases that can be downloaded from internet in runtime. Is there any way to execute `db push` programaticaly before create connection?

**Reply by ludralph** [reply]
@pvmnsd, You can try writing a script  using the `exec` function of Node.js `child_process` module. Something along these lines
```js
const { exec } = require('child_process');
exec(`npx prisma db push`, (error, stdout, stderr) => {
  if (error) {
    console.error(`Database push error: ${error.message}`);
    return;
  }
  if (stderr) {
    console.error(`Database push stderr: ${stderr}`);
    return;
  }
  console.log(`Database push stdout: ${stdout}`);
});
```

**Reply by ludralph** [reply]
Hi @pvmnsd 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by d-kurchenko** [reply]
@ludralph This is not an option, since the PC running the program does not necessarily have `npx prisma`, nor does it know anything about schemas.
